ghcide/src/Development/IDE/Plugin/Test.hs

```diff
@@ -50,10 +48,12 @@ data TestRequest
 newtype WaitForIdeRuleResult = WaitForIdeRuleResult { ideResultSuccess::Bool}
     deriving newtype (FromJSON, ToJSON)
 
-plugin :: P.Plugin c
-plugin = def {
-    P.pluginRules = return (),
-    P.pluginHandlers = requestHandler (SCustomMethod "test") testRequestHandler'
+plugin :: PluginDescriptor IdeState
+plugin = (defaultPluginDescriptor "test") {
+    pluginHandlers = mkPluginHandler (SCustomMethod "test") (\st pId ->
+        if pId == "test"
```

Comment:
Why do we need check `pId` here? I think it should always be `"test"` which we pass to `defaultPluginDescriptor`.

Reply:
That's a good point, hls-plugin-api already takes care of this